### PR TITLE
Bug 1832268: fix pod ring label for failed pod

### PR DESCRIPTION
**Pull request by nemesis09**
Fixes:
https://issues.redhat.com/browse/ODC-2819

Analysis / Root cause:
podring label displays incorrect text for failed pod ( titile: "0", subtitle: "scaling to 1")

Solution Description:
show failed pod in title text of podring label

Screens:
Screenshot from 2020-05-06 17-47-55

Test Coverage:
Screenshot from 2020-05-06 17-50-03

Browser Conformance:
- [x] Firefox
- [x] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by nemesis09** [comment]
The knative service shows no revisions/deployments for failed pods.
cc @serenamarie125

**Comment by debsmita1** [comment]
Could you please add a test for the util `getFailedPods` ?

**Comment by nemesis09** [comment]
> Could you please add a test for the util `getFailedPods` ?

added test for `getFailedPods`

**Comment by serenamarie125** [comment]
> The knative service shows no revisions/deployments for failed pods.
cc @serenamarie125 @beaumorley

I've added an AC in this [Epic](https://issues.redhat.com/browse/ODC-3812) around this issue @nemesis09, thanks for finding it!

File: frontend/packages/console-shared/src/utils/pod-ring-utils.ts [hunk 1]
```diff
@@ -73,6 +74,18 @@ const isPendingPods = (
...
+
+export const getFailedPods = (pods: ExtPodKind[]): number => {
+  let failedPods = 0;
+  pods &&
+    pods.forEach((element) => {
+      if ([AllPodStatus.CrashLoopBackOff, AllPodStatus.Failed].includes(getPodStatus(element))) {
+        failedPods += 1;
+      }
+    });
+  return failedPods;
```
Comment (sahil143) [inline frontend/packages/console-shared/src/utils/pod-ring-utils.ts:86]:
```suggestion
  return pods.reduce((acc, currValue) => {
        if ([AllPodStatus.CrashLoopBackOff, AllPodStatus.Failed].includes(getPodStatus(currValue))) {
        acc += 1;
      }
      return acc;
  }, 0);
```

Comment (nemesis09) [reply frontend/packages/console-shared/src/utils/pod-ring-utils.ts]: updated code to use reduce instead of forEach

**Pull request merged**